### DISCUSSION TITLE
Multiple ConnectionShutdown

### DISCUSSION THREAD

**Discussion by supahero1** [microsoft/msquic#3541] [Q&A] [answered]
Is it alright for me to call ConnectionShutdown on a connection multiple times in a row? For instance every time a stream is closed. I tested it and it doesn't seem to have any side effects. I'm not using the silent flag, but if it has any side effects, it would be cool to know too.
If it's alright, but still sends multiple close packets, then maybe it's not really alright.

**Comment by nibanks** [comment, marked as answer]
Yes, so long as you haven't called ConnectionClose on any other thread in parallel. MsQuic will generally ignore the duplicates.
